### PR TITLE
Fix equivalence key for Populate-Switch

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/44634

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @mavasani another case of https://github.com/dotnet/roslyn/issues/44553. 

This is the second case i've found of this happening and it's now worrying me.  Did we break something here at some point?  These features definitely worked in the past.  However, it seems like we didn't use to merge things with a null-equivalence key, but now we do.

Given that, there's a need to actually go through and audit our fixes (or fixup the test harness) so that we can find out if there are other features which now are no longer working properly.

File: src/Analyzers/Core/CodeFixes/PopulateSwitch/AbstractPopulateSwitchCodeFixProvider.cs [hunk 1]
```diff
@@ -222,7 +222,7 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         private class MyCodeAction : CustomCodeActions.DocumentChangeAction
         {
             public MyCodeAction(string title, Func<CancellationToken, Task<Document>> createChangedDocument)
-                : base(title, createChangedDocument)
+                : base(title, createChangedDocument, title)
```
Comment (mavasani) [inline src/Analyzers/Core/CodeFixes/PopulateSwitch/AbstractPopulateSwitchCodeFixProvider.cs:225]: Can you check the FixAll behavior before and after this fix? I am wondering how it was working if same equivalence key was used for multiple registered code actions for same diagnostic.

Comment (CyrusNajmabadi) [reply src/Analyzers/Core/CodeFixes/PopulateSwitch/AbstractPopulateSwitchCodeFixProvider.cs:225]: We have tests for this in PopulateSwitchStatementTests_FixAllTests.cs.  I imagine it wasn't working properly either.  Possibly another issue with the testing harness?

**Comment by CyrusNajmabadi** [comment]
@mavasani can this go in?

**Pull request merged**